### PR TITLE
Add autodeploy to gh-pages

### LINKED ISSUE #72: published spec text is out of date
@dminor You can use a GitHub action to automatically build and publish to `gh-pages` so this doesn't happen:

https://github.com/tc39/proposal-iterator-sequencing/blob/main/.github/workflows/deploy.yml

Let me know if you want me to send a PR. You'll have to also change the repo settings to deploy pages from the `gh-pages` branch instead of `main`.

### PR DESCRIPTION
**Pull request by dminor**
Fixes #72.

### CODE REVIEW

**Comment by dminor** [comment]
@michaelficarra If you have some time, it would be great to get your feedback on this

**Review by michaelficarra — APPROVED** [review]
Yep this looks great! You'll also have to change the repo settings to deploy from `gh-pages` instead of `main`.

image

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  deploy:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v4
+    - uses: actions/setup-node@v4
+      with:
+        node-version: 20
+    - run: >
+        npm run build
```
Comment (michaelficarra) [inline .github/workflows/deploy.yml:20]: Actually I think you might have to do an `npm install` here first.

Comment (bakkot) [reply .github/workflows/deploy.yml:20]: `npm ci --no-audit` ideally.

**Pull request merged**